src/plugins/intel_cpu/src/mlas/sgemm.hpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+/**
+ * @brief  Computes the length in bytes for the packed matrix B buffer(SGEMM).
+ *
+ * @param N       Supplies the number of columns of matrix B.
+ * @param K       Supplies the number of rows of matrix B.
+ * @return        bytes of the packing buffer
+ */
+size_t mlas_sgemm_pack_get_size(const int64_t N, const int64_t K);
+
+/**
+ * @brief  Packs the contents of matrix B
+ *
+ * @param transb  T for transpose B, N for none-tranpose B
+ * @param N       Supplies the number of columns of matrix B and matrix C.
+ * @param K       Supplies the number of columns of matrix A and the number
+                  of rows of matrix B.
+ * @param ldb     Supplies the first dimension of matrix B.
+ * @param src     Supplies the address of matrix B
+ * @param dst     Supplies pointer to prePacked B buffer
+ */
+void mlas_sgemm_pack(const char* transb,
+                     const int64_t N,
+                     const int64_t K,
+                     const int64_t ldb,
+                     const float* src,
+                     float* dst);
+
+/**
+ * @brief  SGEMM
```

Comment:
1. Do I correctly understand that difference between `mlas_sgemm` and `mlas_sgemm_compute` is B matrix layout? If yes lets directly state in the description that the first one works with planar B matrix, while the second one with prepacked via `mlas_sgemm_pack` function. 
2.  Lets add `thread_num` param handling to `mlas_sgemm_compute` to have aligned API.

Reply:
Yes the difference between `mlas_sgemm` and `mlas_sgemm_compute` is whether B matrix is repakced. Let me comment it clearly.